src/linquantarrays.jl

```diff
@@ -10,84 +13,164 @@ Base.size(QA::LinQuantArray) = size(QA.A)
 Base.getindex(QA::LinQuantArray,i...) = getindex(QA.A,i...)
 Base.eltype(Q::LinQuantArray{T,N}) where {T,N} = T
 
-"""Quantise an array linearly into a LinQuantArray."""
-function LinQuantization(::Type{T},A::AbstractArray) where {T<:Unsigned}
+"""
+
+        LinQuantization(::Type{T}, A::AbstractArray; extrema::Tuple = extrema(A)) where {T<:Integer}
+
+Quantise an array linearly into a LinQuantArray.
+
+# Arguments
+- `T`: the type of the quantised array
+- `A`: the array to quantise
+- `extrema`: the minimum and maximum of the range, defaults to `extrema(A)`.
+
+# Returns
+- a LinQuantArray{T} with the quantised array and the minimum and maximum of the original range.
+"""
+
+function LinQuantization(
+    ::Type{T},
+    A::AbstractArray,
+    extrema::Nothing
+) where {T<:Integer}
     all(isfinite.(A)) || throw(DomainError("Linear quantization only in (-∞,∞)"))
```

Comment:
Looking at this here, we should probably first have the `extrema` call, then simply check whether min and max are finite (they wouldn't be if any element in `A` isn't finite). The problem with `all(isfinite.(A))` is that it'll allocate a bitarray and reduce that again to a single boolean. I know I wrote that, but it's not what we should do 😆